worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/QueueHandler.java

```diff
@@ -50,7 +51,13 @@ public abstract class QueueHandler implements Trimable, Runnable {
             Settings.settings().QUEUE.PARALLEL_THREADS,
             new FaweForkJoinWorkerThreadFactory("FAWE Fork Join Pool Primary - %s"),
             null,
-            false
+            false,
+            Settings.settings().QUEUE.PARALLEL_THREADS,
```

Comment:
The previous maximum pool size would have been the default of (what seems to be) `0x7fff` (32767) - is this change made on purpose? The secondary pool would have this default value

Reply:
Yes, I didn't find any better solution to this (besides e.g. using a semaphore per edit), but otherwise, waiting for chunk loads will increase the allowed numbers of threads, and if there are more tasks enqueued, this will cause more threads to be started. Those threads will process tasks and might end up waiting for a chunk to load, resulting in even more threads. I often had around as much threads as chunks in my tests, so this hard limit looked like a good solution. As we now allow processing multiple edits on one thread, this also won't cause other edits to be stalled.

I'm open for alternatives though.

Reply:
I wonder if it's worth replacing the Future extension in the chunk flow with ForkJoinTask - we can then look at ways to better implement the chunk async await (alongside the earlier PR to not await chunk load when calling)

Reply:
I suppose for this PR it's a bit out of scope, but I don't think such a change would be a "breaking"/v3 change anyway as none of this is particularly API